epictrack-web/src/components/workPlan/event/EventForm.tsx

```diff
@@ -193,8 +191,8 @@ const EventForm = ({ onSave, event, isFormFieldsLocked }: EventFormProps) => {
       data.notes = notes;
       // setSubmittedEvent(data);
       const showConfirmDialog =
-        (event === undefined && data.actual_date != "") ||
-        (event != null && event.actual_date == null && data.actual_date != "");
+        (event === undefined && !!data.actual_date) ||
+        (event != null && event.actual_date == null && !!data.actual_date);

```

Comment:
Can we leverage JavaScripts syntax. instead of event !=null we can do !event. and we can chain. instead of event !=null && event.actual_date == null we can do !event?.actual_date. And we don't need !! if data.actual_date exists it will be truthy so just && data.actual_date

Reply:
Also any chance we can create constants for the conditions just to add some meaning.

example: const descriptiveName = event === undefined && !!data.actual_date